pkg/storaged/format-dialog.jsx

```diff
@@ -399,14 +403,23 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
         update: function (dlg, vals, trigger) {
             if (trigger == "at_boot")
                 dlg.set_options("at_boot", { explanation: mount_explanation[vals.at_boot] });
+            else if (trigger == "type") {
+                mount_type = dlg.get_value("type");
+                if (mount_type == "empty") {
+                    Object.assign(this.Action, { Variants: null, Title: _("Format") });
+                } else {
+                    Object.assign(this.Action, { Variants: [action_variant], Title: action_title });
+                }
+            }
+            dlg.update_footer();
```

Comment:
Should not dlg.update_footer*() happen only in the body of the `else if`?

Otherwise looks good to me as an approach. @mvollmer do you have maybe an opinion?

@MahmoudHamdy02 could be really great if you could extend the test for this. I guess we could extend this https://github.com/cockpit-project/cockpit/blob/main/test/verify/check-storage-format#L89

If you find it hard - I can also help you with this.

Reply:
> Should not dlg.update_footer*() happen only in the body of the `else if`?

Yup, fixing it now!
I will look into the test and let you know if I have any trouble